### PR TITLE
12.0 afc 1741

### PR DESCRIPTION
**Pull request by SebastianGieser-Vertel**

### CODE REVIEW

File: partner_tlr_update/models/res_partner.py [hunk 1]
```diff
@@ -165,23 +161,35 @@ def parse_xml_tjansteleverantor_data(self, xml):
...
     @api.multi
-    def update_contact_person(self, xml, parent_id, address_id):
+    def get_contract_nrs(self, xml, service_nr = 26):
+        contract_nrs = []
+        contracts = xml.findall(".//ns107:avtalLista", namespaces={'ns107': "http://arbetsformedlingen.se/datatyp/tjansteleverantor/tjansteleverantor/v15"})
+        if contracts:
+            _logger.info("contracts found")
+            for contract in contracts:
+                service_id = contract.find(".//ns62:tjanstId", namespaces={'ns62':'http://arbetsformedlingen.se/datatyp/tjansteleverantor/avtal/v15'})
+                if service_id is not None:
```
Comment (nils-vertel) [inline partner_tlr_update/models/res_partner.py]: Can't this be replaced with if service_id: ?

File: partner_tlr_update/models/res_partner.py [hunk 2]
```diff
@@ -165,23 +161,35 @@ def parse_xml_tjansteleverantor_data(self, xml):
...
+                    _logger.info("service number %s found, service number wanted: %s" % (service_id.text,service_nr))
+                    if service_id.text == str(service_nr):
+                        _logger.info("service number %s found" % service_nr)
+                        contract_id = contract.find(".//ns62:avtalId", namespaces={'ns62':'http://arbetsformedlingen.se/datatyp/tjansteleverantor/avtal/v15'})
+                        if contract_id is not None:
```
Comment (nils-vertel) [inline partner_tlr_update/models/res_partner.py]: Same here, is it important that its not None? or can this be replaced with if contract_id

File: partner_tlr_update/models/res_partner.py [hunk 3]
```diff
@@ -213,7 +221,12 @@ def update_contact_person(self, xml, parent_id, address_id):
                 })
 
     @api.multi
-    def update_subsidiary(self, xml):
+    def update_subsidiary(self, xml, contract_nrs):
+        contract_id = xml.find(".//ns64:avtalId",
+                                 namespaces={'ns64': 'http://arbetsformedlingen.se/datatyp/tjansteleverantor/utforandeverksamhet/v15'})
+        if contract_id is None or not contract_id.text in contract_nrs:
```
Comment (nils-vertel) [inline partner_tlr_update/models/res_partner.py]: Could be replaced with: If not (contract_id and contract_id.text in contract_nrs):

File: partner_tlr_update/models/res_partner.py (hunk 1 above)
Comment (nils-vertel) [inline partner_tlr_update/models/res_partner.py]: This if chain is pretty deeply nested, consider compacting it a bit (you will lose some of the logging)

**Review by nils-vertel — CHANGES_REQUESTED** [review]
Running a Linter on the code gives 63 problems, please check those.
There is quite a lot of info logging, check on the logs and see if they are necessary, maybe downgrade them to debug.

File: partner_tlr_update/models/res_partner.py (hunk 1 above)
Comment (SebastianGieser-Vertel) [reply partner_tlr_update/models/res_partner.py]: No, it throws warnings about how it's not going to work in the future, it says to use either len(elem) or elem not None.

**Pull request merged**